source/common/quic/quic_stat_names.cc

```diff
@@ -0,0 +1,42 @@
+#include "common/quic/quic_stat_names.h"
+
+namespace Envoy {
+namespace Quic {
+
+QuicStatNames::QuicStatNames(Stats::SymbolTable& symbol_table)
+    : stat_name_pool_(symbol_table), symbol_table_(symbol_table),
+      downstream_(stat_name_pool_.add("downstream")), upstream_(stat_name_pool_.add("upstream")),
+      from_self_(stat_name_pool_.add("self")), from_peer_(stat_name_pool_.add("peer")) {
+  // Preallocate most used counters
+  connectionCloseStatName(quic::QUIC_NETWORK_IDLE_TIMEOUT);
+}
+
+void QuicStatNames::incCounter(Stats::Scope& scope, const Stats::StatNameVec& names) {
+  Stats::SymbolTable::StoragePtr stat_name_storage = symbol_table_.join(names);
```

Comment:
maybe comment an example counter name?

Reply:
done adding comments in the method declaration.